sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousMetricStorage.java

```diff
@@ -100,22 +123,21 @@ static <T extends PointData, U extends ExemplarData> AsynchronousMetricStorage<T
    * Record callback measurement from {@link ObservableLongMeasurement} or {@link
    * ObservableDoubleMeasurement}.
    */
+  @SuppressWarnings("UnnecessaryDefaultInEnumSwitch")
   void record(Measurement measurement) {
     Context context = Context.current();
     Attributes processedAttributes = attributesProcessor.process(measurement.attributes(), context);
     long start =
         aggregationTemporality == AggregationTemporality.DELTA
             ? registeredReader.getLastCollectEpochNanos()
             : measurement.startEpochNanos();
-    measurement =
-        measurement.hasDoubleValue()
-            ? Measurement.doubleMeasurement(
-                start, measurement.epochNanos(), measurement.doubleValue(), processedAttributes)
-            : Measurement.longMeasurement(
-                start, measurement.epochNanos(), measurement.longValue(), processedAttributes);
+
+    measurement = measurement.withAttributes(processedAttributes).withStartEpochNanos(start);
+
     recordPoint(processedAttributes, measurement);
   }
 
+  @SuppressWarnings("UnnecessaryDefaultInEnumSwitch")
```

Comment:
These `@SuppressWarnings` annotations are not needed.

Reply:
Remove all 4 of those